src/bridge/cockpitchannel.c

```diff
@@ -386,6 +391,10 @@ cockpit_channel_set_property (GObject *object,
       g_return_if_fail (self->priv->capabilities == NULL);
       self->priv->capabilities = g_value_dup_boxed (value);
       break;
+    case PROP_GROUP_OBJECTS:
+      g_return_if_fail (self->priv->group_objects == NULL);
+      self->priv->group_objects = g_value_dup_boxed (value);
```

Comment:
Totally agree with you. I've put the `g_return_if_fail` (like the other construct-only properties) so that one at least starts wondering when reading it. I'd love if `GObject` had a better solution to this.